app/main.py

```diff
@@ -0,0 +1,474 @@
+})
+
+# Device List Data Model
+LIST_OF_DEVICES = API.model('ListedDevices', {
+    'id': fields.String(required=True, description='The device ID'),
+    'device': fields.Nested(DEVICE, description='The device'),
+})
+
+HEARTBEAT = API.model('Heartbeat', {
+    'heartbeat': fields.String(required=False, description='Device heartbeat')
+})
+
+
+class Redis(object):
+    """Class for manipulating redis client."""
+
+    def __init__(self):
+        """Initialize Redis Object."""
+        if 'REDIS_HOST' in os.environ:
+            REDIS_HOST = os.environ['REDIS_HOST']
+            if 'REDIS_PASSWORD' in os.environ:
+                REDIS_PASSWORD = os.environ['REDIS_PASSWORD']
+                logging.info("Redis password found %s", REDIS_PASSWORD)
+                self.redis = redis.Redis(
+                    host=REDIS_HOST, port=6379, db=0, password=REDIS_PASSWORD)
+            else:
+                logging.info(
+                    "No redis password found, initializing without auth")
+                self.redis = redis.Redis(host=REDIS_HOST, port=6379, db=0)
+        else:
+            try:
+                raise Exception
+            except Exception:
+                logging.error('REDIS_HOST not set.')
+                sys.exit("!!!Exiting. Please set REDIS_HOST in env & restart.")
+
+    def health(self):
+        """Return a ping response from redis client."""
+        return self.redis.ping()
+
+    def get(self, key):
+        """Get device from redit if it exists, otherwise load with DEFAULT."""
+        with self.redis.pipeline() as pipe:
+            while True:
+                try:
+                    pipe.watch(key)
+                    if pipe.exists(key) == 0:
+                        logging.warning("get: no device")
+                        pipe.multi()
+                        pipe.hmset(key, DEFAULT)
+                        pipe.execute()
+                        device = DEFAULT
+                        break
+                    else:
+                        device = pipe.hgetall(key)
+                        # Initialising empty dictionary
+                        new_dict = {}
+
+                        # Convert dictionary items from bytes to strings
+                        for dkey, value in device.items():
+                            new_dict[dkey.decode(
+                                "utf-8")] = value.decode("utf-8")
+                        device = new_dict
+                        logging.debug("get: device found: %s", str(device))
+                        break
+                except WatchError:
+                    logging.warning("get: watcherror, trying again")
+                    continue
+                except redis.ConnectionError:
+                    logging.exception("Connection error")
+                    return "Error: error connectiong to Redis host"
+
+                except Exception:
+                    logging.exception("Unexpected error")
```

Comment:
Right now for connection errors you're returning a string. I haven't checked everywhere that uses this, but I suspect that it might be surprising to call redis.get(....) and receive a string when you're expecting an object.

Reply:
This should be address in new version in master.